src/engraving/style/scorestylepreset.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef MU_ENGRAVING_SCORESTYLEPRESET_H
+#define MU_ENGRAVING_SCORESTYLEPRESET_H
+
+#include "global/iglobalconfiguration.h"
+#include "notation/inotationconfiguration.h"
```

Comment:
This file contains `#include <QColor>`, which causes the CI_Without_Qt check to [fail](https://github.com/musescore/MuseScore/actions/runs/9566693200/job/26372587841?pr=23048#step:5:331).

That's a clue that this is the wrong approach here. You shouldn't need to use Qt functionality in the engraving module.

One solution might be to return an `MStyle` object directly, instead of returning a path to a style file. That would enable you to call `DefaultStyle::defaultStyle()`.

Otherwise, you'll need another way to get the file path to the default style.

Reply:
Probably your `getStyleFilePath()` function should behave more like [DefaultStyle::resolveStyleDefaults()](https://github.com/musescore/MuseScore/blob/1c28ffe8f79b3f663af481d81dd6cfcb886fe971/src/engraving/style/defaultstyle.cpp#L119-L154).

The first time `DefaultStyle::resolveStyleDefaults()` is called with a certain argument, it creates an `MStyle` object from a style file. If it gets called again in future with the same argument, it just returns the same `MStyle` object it created on the first call. So the actual style files only get accessed once.

Reply:
Another possibility is to use `IEngravingConfiguation` instead of `INotationConfiguration`. `IEngravingConfiguration` namely also has a `defaultStyleFilePath` method. (In fact, the one from `INotationConfiguration` _uses_ `IEngravingConfiguration`.)

In general, you shouldn't really ever include `notation` files in the `engraving` module. This is to avoid circular dependencies. Instead, the dependencies between modules should be seen as a hierarchy, and `engraving` is near the top of that hierarchy, meaning that it should not depend on most modules.

Reply:
I see, yes that makes sense. Let me try it out